optimum/habana/transformers/generation/utils.py

```diff
@@ -170,6 +170,50 @@ def _update_model_kwargs_for_generation(
 
         return model_kwargs
 
+    def _prepare_decoder_attention_mask(
+        self,
+        max_steps: int,  #current stopping criteria
+        batch_size: int,
+        pad_token_id: int,
+        device: str,
+        dtype: str = torch.int64,
+    ) -> torch.Tensor:
+        x = torch.zeros((batch_size, max_steps), device=device, dtype=dtype)
+        return x.index_fill(1, torch.tensor([0]), pad_token_id)    # First the position with pad_token_id
```

Comment:
I think it's equivalent to what is done [here](https://github.com/huggingface/optimum-habana/blob/7204bbbb1e28e11c3b99ff6e3b5a2bc9930886dc/optimum/habana/transformers/generation/utils.py#L364) in #293 no?